subgraphs/erc20/schema.graphql

```diff
@@ -0,0 +1,292 @@
+# Subgraph Schema: ERC20 Tokens
+# Version: 1.0.0
+
+#############################
+####### Token Metadata ######
+#############################
+
+type Token @entity {
+  " Smart contract address of the token "
+  id: ID!
+
+  " Name of the token, mirrored from the smart contract "
+  name: String!
+
+  " Symbol of the token, mirrored from the smart contract "
+  symbol: String!
+
+  " The number of decimal places this token uses, default to 18 "
+  decimals: Int!
+
+  " If token transfers are paused, derived from paused event "
+  isPaused: Boolean
+
+  " Number of unique token holders "
+  holderCount: BigInt!
+
+  " Total number of token transfer transactions "
+  transferCount: BigInt!
+
+  " Total number of token mint transactions "
+  mintCount: BigInt!
+
+  " Total number of token burn transactions "
+  burnCount: BigInt!
+
+  " Total token supply "
+  totalSupply: BigDecimal!
+
+  " Total token burned "
+  totalBurned: BigDecimal!
+
+  " Total token minted "
+  totalMinted: BigDecimal!
+
+  " List of token transfers "
+  transfers: [Transfer!]! @derivedFrom(field: "token")
+
+  " Token holder's balance "
+  holdersBalance: [AccountBalance!]! @derivedFrom(field: "token")
+
+  ##### Snapshots #####
+
+  " Daily usage metrics for this token "
+  dailyUsageMetrics: [UsageMetricsDailySnapshot!]! @derivedFrom(field: "token")
+
+  " Hourly usage metrics for this token "
+  hourlyUsageMetrics: [UsageMetricsHourlySnapshot!]! @derivedFrom(field: "token")
+
+  dailyTokenSnapshot: [TokenDailySnapshot!]! @derivedFrom(field: "token")

```

Comment:
since there is one field in here, how about we combine it with usage metrics and call them `tokenSnapshot`? e.g. `tokenDailySnapshot` and `tokenHourlySnapshot`

Reply:
yeah, we can do that